src/components/MultiHopTrade/components/TradeInput/components/ConfirmSummary.tsx

```diff
@@ -115,6 +117,13 @@ export const ConfirmSummary = ({
     buyAsset.chainId,
     buyAsset.assetId,
   ])
+  const displayManualAddressEntry = useMemo(() => {
+    // If accounts have finished loading, *and* there is no runtime chain support for the buy asset, display manual address entry
+    if (!walletSupportsBuyAssetChain) return !isAccountsLoading
+
+    return disableThorNativeSmartContractReceive === true
+  }, [walletSupportsBuyAssetChain, disableThorNativeSmartContractReceive, isAccountsLoading])
```

Comment:
nitpick suggestion - more explicit and readable conditions imo. can make the wallet + sc check more concise if you want, but most important imo is the early accounts loading return.
```suggestion
  
  const displayManualAddressEntry = useMemo(() => {
    if (isAccountsLoading) return false

    if (!walletSupportsBuyAssetChain) return true
    if (disableThorNativeSmartContractReceive) return true

    return false
  }, [walletSupportsBuyAssetChain, disableThorNativeSmartContractReceive, isAccountsLoading])
```

Reply:
Fixed in [e72bded](https://github.com/shapeshift/web/pull/7629/commits/e72bded769397fe997646cd23ed7ff15a20d7bc6)